managed/services/inventory/inventory_metrics.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package inventory
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm/managed/models"
+)
+
+const (
+	requestTimeout              = 3 * time.Second
+	serviceEnabled      float64 = 1
+	prometheusNamespace         = "pmm_managed"
+	prometheusSubsystem         = "inventory"
+)
+
+type Metric struct {
+	labels []string
+	value  float64
+}
+
+//goland:noinspection GoNameStartsWithPackageName
+type InventoryMetrics struct {
+	db       *reform.DB
+	registry agentsRegistry
+}
+
+//goland:noinspection GoNameStartsWithPackageName
+type InventoryMetricsCollector struct {
+	mAgentsDesc   *prom.Desc
+	mNodesDesc    *prom.Desc
+	mServicesDesc *prom.Desc
+
+	metrics inventoryMetrics
+
+	mutex sync.Mutex
+}
+
+func NewInventoryMetrics(db *reform.DB, registry agentsRegistry) *InventoryMetrics {
+	return &InventoryMetrics{
+		db:       db,
+		registry: registry,
+	}
+}
+
+func NewInventoryMetricsCollector(metrics inventoryMetrics) *InventoryMetricsCollector {
+	return &InventoryMetricsCollector{
+		mAgentsDesc: prom.NewDesc(
+			prom.BuildFQName(prometheusNamespace, prometheusSubsystem, "agents"),
+			"Inventory Agent",
+			[]string{"agent_id", "agent_type", "service_id", "node_id", "pmm_agent_id", "disabled", "version"},
+			nil),
+		mNodesDesc: prom.NewDesc(
+			prom.BuildFQName(prometheusNamespace, prometheusSubsystem, "nodes"),
+			"Inventory Node",
+			[]string{"node_id", "node_type", "node_name", "container_name"},
+			nil),
+		mServicesDesc: prom.NewDesc(
+			prom.BuildFQName(prometheusNamespace, prometheusSubsystem, "services"),
+			"Inventory Service",
+			[]string{"service_id", "service_type", "node_id"},
+			nil),
+
+		metrics: metrics,
+	}
+}
+
+func (i *InventoryMetrics) GetAgentMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbAgents, dbAgentsError := models.FindAgents(tx.Querier, models.AgentFilters{})
+
+		if dbAgentsError != nil {
+			return dbAgentsError
+		}
+
+		for _, agent := range dbAgents {
+			disabled := "0"
+			connected := float64(0)
```

Comment:
why in one case it is string, while int in the other

Reply:
because `connected` is a value, but `disabled` is label (which should be as string)